File: lib-jvm/src/main/java/com/n8cats/share/data/PlayerId.java
package com.n8cats.share.data;
public class PlayerId {
  @SuppressWarnings("unused") public PlayerId() {
  }
  public PlayerId(int id) {
    this.id = id;
  }
  public int id;
  public int hashCode() {
    return id;
  }
  public boolean equals(Object o) {
    return o != null && (o == this || o.getClass() == PlayerId.class && ((PlayerId) o).id == id);
  }
  public String toString() {
    return String.valueOf(id);
  }
}

File: lib-jvm/src/main/java/com/n8cats/share/data/PlayerId.kt
package com.n8cats.share.data

class PlayerId {
  var id:Int = 0

  constructor() {}
  constructor(id:Int) {
    this.id = id
  }

  override fun hashCode():Int {
    return id
  }

  override fun equals(o:Any?):Boolean {
    return o!=null&&(o===this||o.javaClass==PlayerId::class.java&&(o as PlayerId).id==id)
  }

  override fun toString():String {
    return id.toString()
  }
}
